components/apps_form/apps_form_field/apps_form_select_field.tsx

```diff
@@ -123,10 +201,14 @@ export default class AppsFormSelectField extends React.PureComponent<Props, Stat
         const {field, label, helpText} = this.props;
 
         let selectComponent;
-        if (field.type === 'dynamic_select') {
+        if (field.type === AppFieldTypes.DYNAMIC_SELECT) {
```

Comment:
Probably best to use a `switch` statement for these since we have 4 types now

Reply:
I'm also a fan of not using switch when we can use a structure, something like:

```javascript
DYNAMIC = 'dynamic'
CHANNEL = 'channel'

const availableRenders = {
    DYNAMIC: renderThis,
    CHANNEL: renderThat,
}

availableRenders[DYNAMIC]()
```

Reply:
In using it in this way, wouldn't we have to define the renderers for all types which are still not used like `AppFieldTypes.Text`, etc; for which it is null currently
@javaguirre